python/mlcroissant/mlcroissant/_src/core/versions.py

```diff
@@ -19,10 +19,9 @@ def cast_version(version: Any) -> str | None:
     if isinstance(version, str):
         numbers = version.split(".")
         are_not_all_numbers = any(not number.isnumeric() for number in numbers)
-        if len(numbers) != 3 or are_not_all_numbers:
+        if are_not_all_numbers:
             raise WarningException(
-                f"Version doesn't follow MAJOR.MINOR.PATCH: {version}. For more"
-                " information refer to: https://semver.org/spec/v2.0.0.html"
+                f"Version contains non-numeric characters: {version}."
```

Comment:
Keeping the warning for non-numeric versioning. Reasonable? Or shall I remove this as well?

Reply:
We could fully implement the [semantic versioning standard](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions).

[`semver`](https://pypi.org/project/semver) is a python library that does exactly this. However it seems we only need to check the compatibility with the standard, so we could also just reuse their [regular expression](https://github.com/python-semver/python-semver/blob/c47774db88386f7db79e5ec9fe3672c07ed7c063/src/semver/version.py#L81-L104).

Reply:
Also, I think that it's worth mentioning the semantic versioning standard: https://semver.org/spec/v2.0.0.html, and that minor and patch can be omitted.